tracer/src/Datadog.Trace/Ci/CiEnvironment/GitInfo.cs

```diff
@@ -0,0 +1,184 @@
+    /// <returns>Git info</returns>
+    public static IGitInfo GetFrom(string folder)
+    {
+        List<string>? errors = null;
+        foreach (var provider in _gitInfoProviders)
+        {
+            // Try to load git metadata from the folder
+            if (provider.TryGetFrom(folder, out var gitInfo) && gitInfo != null)
+            {
+                return gitInfo;
+            }
+
+            if (gitInfo != null)
+            {
+                errors ??= new List<string>();
+                errors.AddRange(gitInfo.Errors);
+            }
+        }
+
+        // Return the partial gitInfo instance with the initial source root
+        var value = new GitInfo { SourceRoot = new DirectoryInfo(folder).Parent?.FullName };
+        if (errors != null)
+        {
+            value.Errors.AddRange(errors);
+            Log.Warning("{Value}", string.Join(Environment.NewLine, value.Errors));
+        }
+
+        return value;
+    }
+
+    /// <summary>
+    /// Gets a GitInfo from the current folder or assembly attribute
+    /// </summary>
+    /// <returns>Git info</returns>
+    public static IGitInfo GetCurrent()
+    {
+        List<string>? errors = null;
+        var baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+        var gitDirectory = GetParentGitFolder(baseDirectory) ?? GetParentGitFolder(Environment.CurrentDirectory);
+        if (gitDirectory != null)
+        {
+            foreach (var provider in _gitInfoProviders)
+            {
+                if (provider.TryGetFrom(gitDirectory.FullName, out var gitInfo) && gitInfo != null)
+                {
+                    return gitInfo;
+                }
+
+                if (gitInfo != null)
+                {
+                    errors ??= new List<string>();
+                    errors.AddRange(gitInfo.Errors);
+                }
+            }
+        }
+
+        var value = new GitInfo { SourceRoot = gitDirectory?.Parent?.FullName };
+        if (errors != null)
+        {
+            value.Errors.AddRange(errors);
+            Log.Warning("{Value}", string.Join(Environment.NewLine, value.Errors));
+        }
+
+        return value;
+    }
+
+    public static DirectoryInfo? GetParentGitFolder(string? innerFolder)
+    {
+        if (string.IsNullOrEmpty(innerFolder))
+        {
+            return null;
+        }
+
+        var dirInfo = new DirectoryInfo(innerFolder);
+        while (dirInfo != null)
+        {
+            var gitDirectories = dirInfo.GetDirectories(".git");
+            if (gitDirectories.Length > 0)
+            {
+                foreach (var gitDir in gitDirectories)
+                {
+                    if (gitDir.Name == ".git")
+                    {
+                        return gitDir;
+                    }
+                }
+            }
+
+            dirInfo = dirInfo.Parent;
+        }
+
+        return null;
+    }
+
+    internal static void SetGitInfoProviders(params IGitInfoProvider[] providers)
```

Comment:
Just for testing, right? Maybe make that clear in the name?

Reply:
not in use, removed in [d107ce2](https://github.com/DataDog/dd-trace-dotnet/pull/6259/commits/d107ce21648aec65ae5ba0a696ec0c908d74e0c6)